packages/SwingSet/src/devices/mailbox-src.js

```diff
@@ -90,23 +89,23 @@ export function buildRootDeviceNode(tools) {
       try {
         endowments.add(`${peer}`, Nat(msgnum), `${body}`);
       } catch (e) {
-        throw new Error(`error in add: ${e}`);
+        assert.fail(X`error in add: ${e}`);
```

Comment:
Incidentally, this re-throwing of a stringified Error was originally there to protect against cross-Realm contamination, where the kernel-Realm code could extract a full-powered `Function` constructor from the primal-Realm `Error` object.

I'm glad we don't need that anymore.

Reply:
Should I remove the extra layer?

Reply:
not now, not worth the thinking time to make sure it's safe, but let's add a low-priority cleanup ticket to consider that

Reply:
deferred to #2421